### DISCUSSION TITLE
Invalid Digital Signature on install

### DISCUSSION THREAD

**Discussion by Tothiro** [ExistentialAudio/BlackHole#460] [Q&A]
Apologies if this is a smooth brain question, but this is my first github post... 
I was looking to try Blackhole to set up an aggregate but after following the email confirmation link to DL, this legacy Macbook is throwing me an "Invalid Digital Signature" error and telling me to kick rocks.
I'm attempting on a 10.11.6 OS, which appears as it should be supported... am I incorrect in that?

tl:dr:
"BlackHole2ch.v0.2.10.pkg can't be installed because its digital signature is invalid."   - What do?

**Comment by devinroth** [comment]
In short, Apple has left you behind. App signing on Big Sur is no longer compatible with older OS versions.

Yes, BlackHole is compatible but unfortunately the installer is not. The only way for me to fix it is by loading up a Virtual Box with an older version of MacOS. It’s a real pain.

I get contacted about this issues about once a week. Is there enough legacy users out there to go through the hassle? If it’s really worth it to users like you, become a sponsor and buy me some caffeine so I can suck it up, dust off the ol’ Virtual Box, and sign the app in a way that works.

And I’ll have to do it again after every update. Sigh.

Devin

**Reply by mikenye** [reply]
Is this something that could be automated in a github action?

Github hosted runners include macos-10.15, so the caffeine may be better spent writing the workflow so you don't have to do it each update... :-)

Just a thought.

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

**Reply by devinroth** [reply]
That's not a bad idea. Although I think the signature problem started with 10.15. I don't remember for sure. I'll look into it.

**Reply by 3gyptian** [reply]
So as a workaround I installed BlackHole on my newer Catlina Mac no problemo .. and then copied over the driver files to my beater El Captain Mac :

```drwxr-xr-x  0 root   wheel       0  7 Aug 10:07 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/
drwxr-xr-x  0 root   wheel       0  7 Aug 10:06 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/
drwxr-xr-x  0 root   wheel       0  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/_CodeSignature/
drwxr-xr-x  0 root   wheel       0  7 Aug 14:42 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/MacOS/
drwxr-xr-x  0 root   wheel       0  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/Resources/
-rw-r--r--  0 root   wheel    1656  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/Info.plist
-rw-r--r--  0 root   wheel   35149  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/Resources/LICENSE
-rw-r--r--  0 root   wheel    2307  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/Resources/CHANGELOG.md
-rw-r--r--  0 root   wheel    7451  7 Aug 09:46 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/Resources/BlackHole.icns
-rwxr-xr-x  0 root   wheel  279296  7 Aug 14:42 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/MacOS/BlackHole
-rw-r--r--  0 root   wheel    3116  7 Aug 14:42 /Library/Audio/Plug-Ins/HAL/BlackHole2ch.driver/Contents/_CodeSignature/CodeResources
```

And then on the beater Mac I ran: 

```
sudo launchctl kickstart -k system/com.apple.audio.coreaudiod
```


And that seems to have done the trick.   

I guess my question would be how to extract those driver files from the package without having a newer Mac. I know I can unpackage the package via


```
pkgutil --expand BlackHole2ch.v0.2.10.pkg temp
```

So I look in the temp folder and there's a bunch of files I have no idea how to extract the above driver files without using the macos installer utility.

thanks for any insight on this!

**Comment by devinroth** [comment]
You can download the source code and compile  it.

Devin

**Comment by schubidubi** [comment]
You can use **Pacifist** to run the installer: 
https://charlessoft.com
You may get a similar warning as with the MacOS Installer but then simply click "Continue".
Pacifist should work in demo mode indefinitely, it just always takes its time at each launch.
